src/dfx/src/commands/deploy.rs

```diff
@@ -150,26 +161,44 @@ fn display_urls(env: &dyn Environment) -> DfxResult {
 
     if let Some(canisters) = &config.get_config().canisters {
         for (canister_name, canister_config) in canisters {
-            if config
+            let canister_is_remote = config
                 .get_config()
-                .is_remote_canister(canister_name, &network.name)?
-            {
+                .is_remote_canister(canister_name, &network.name)
+                .with_context(|| {
+                    format!(
+                        "Failed to determine if canister {} is remote.",
+                        canister_name
+                    )
+                })?;
```

Comment:
I notice this is repeated (as an example; CanisterIdStore::for_env is similar in this respect).

This is a good example of a place where we aren't actually adding context (at this level) that can't be added by the method being called.  Add to that, that we're actually missing something (the network name), that would be easier to get right if it were done in one place.
```suggestion
                .is_remote_canister(canister_name, &network.name)
                .with_context(|| {
                    format!(
                        "Failed to determine if canister {} is remote for network {}.",
                        canister_name,
                        &network.name
                    )
                })?;
```

Can you think of a good solution to reduce code duplication?